torch_xla/csrc/tensor_util.cpp

```diff
@@ -947,8 +949,13 @@ std::vector<torch::lazy::BackendDataPtr> CreateTensorsData(
       // Shards the input tensors with padding, to split evenly.
       // The execution requires consistent shard sizes, and the zero-padded
       // values should be ignored.
-      std::vector<at::Tensor> local_shards = ShardingUtil::ShardTensor(
-          tensors[i], sharding, local_devices, /*padded=*/true);
+      std::vector<at::Tensor> local_shards =
+          ShardingUtil::ShardTensor(tensors[i], shardings[i], local_devices,
```

Comment:
`shardings[i]` could be nullptr, in which case we want to pass in a REPLICATED sharding spec to ShardTensor. Or, we can handle a null input in this if block: https://github.com/pytorch/xla/blob/8ada333234e376ed217b1937a0ce2da86b832d70/torch_xla/csrc/xla_sharding_util.cpp#L477

Reply:
I'm also curious on the deletion of the null check.

Reply:
This is handled within ShardingUtil::ShardTensor line 485..